packages/ngtools/webpack/src/resource_loader.ts

```diff
@@ -110,17 +111,23 @@ export class WebpackResourceLoader {
       throw new Error('WebpackResourceLoader cannot be used without parentCompilation');
     }
 
-    // Create a special URL for reading the resource from memory
-    const entry =
-      filePath ||
-      (resourceType
-        ? `${containingFile}-${this.outputPathCounter}.${fileExtension}!=!${this.inlineDataLoaderPath}!${containingFile}`
-        : // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-          `angular-resource:${resourceType},${createHash('md5').update(data!).digest('hex')}`);
+    const getEntry = (): string => {
+      if (filePath) {
+        return `${filePath}?${NG_COMPONENT_RESOURCE_QUERY}`;
+      } else if (resourceType) {
+        return (
+          `${containingFile}-${this.outputPathCounter}.${fileExtension}` +
+          `?${NG_COMPONENT_RESOURCE_QUERY}!=!${this.inlineDataLoaderPath}!${containingFile}`
```

Comment:
Nit: These templates are pretty complicated, so I wonder if an example in a comment might make more clear what kind of path we're generating here.